clash-lib/src/Clash/Driver/Bool.hs

```diff
@@ -0,0 +1,34 @@
+{-|
+  Copyright   :  (C) 2024, Martijn Bastiaan
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveAnyClass #-}
+
+module Clash.Driver.Bool where
+
+import Control.DeepSeq (NFData)
+import Data.Hashable (Hashable)
+import GHC.Generics (Generic)
+
+#if MIN_VERSION_ghc(9,4,0)
+import GHC.Data.Bool (OverridingBool(..))
+#elif MIN_VERSION_ghc(9,0,0)
+import GHC.Utils.Misc (OverridingBool(..))
+#else
+import Util (OverridingBool(..))
+#endif
+
+data COverridingBool = CAuto | CNever | CAlways
```

Comment:
Can’t we just name it the same way as GHC? Instead of using this 'C' prefix for both the type and data constructors.

And the name the conversion functions: from/toGHCOverridingBool.

We have a lot of the same names as the GHC code base, so it’s not like this would be the first time we would do this.

Reply:
Done (with slightly different capitalization due to our style guide)

Reply:
But the type constructor still has the 'C' prefix, can't we remove that as well?

Reply:
Darn it, missed that with renaming

Reply:
Done

Reply:
Thanks